src/map.cpp

```diff
@@ -5652,6 +5656,79 @@ void map::draw( const catacurses::window &w, const tripoint &center )
     }
 }
 
+void map::update_terrain_weather()
+{
+    const w_point weatherPoint = *g->weather_precise;
+    if( g->weather == WEATHER_FLURRIES ) {
+        snowfall += 2;
+    } else if( g->weather == WEATHER_SNOW ) {
+        snowfall += 4;
+    } else if( g->weather == WEATHER_SNOWSTORM ) {
+        snowfall += 8;
+    } else {
+        snowfall -= std::max( 0, ( static_cast<int>( weatherPoint.temperature - 32 ) ) );
+    }
+    snowfall = std::max( 0, snowfall );
+    snowfall = std::min( 172800, snowfall );
+}
+
+void map::add_snow()
+{
+    if( g->weather >= WEATHER_FLURRIES && snowfall > 3000 ) {
+        tripoint gp( 0, 0, 0 );
+        int total_grid = 0;
+        int snowfall_settled = 0;
+        int &gx = gp.x;
+        int &gy = gp.y;
+        int snowfactor = std::min( 15000, snowfall );
+        snowfactor = 15001 - snowfactor;
+        for( gx = 0; gx < my_MAPSIZE * SEEX; ++gx ) {
+            for( gy = 0; gy < my_MAPSIZE * SEEY; ++gy ) {
+                total_grid += 1;
```

Comment:
There's no need to increment here. This value can be calculated in compile time outside the loops as `my_MAPSIZE * SEEX * my_MAPSIZE * SEEY`